study_chess/src/main/java/studychess/services/JwtService.java

```diff
@@ -24,7 +24,7 @@ public String generateToken(UserModel user) {
             Algorithm algorithm = Algorithm.HMAC256(secret);
             String token = JWT.create()
                     .withIssuer("studychess")
-                    .withSubject(user.getUsername())
+                    .withSubject(user.getLogin())
```

Comment:
Another option here is to update the getUsername method in UserModel to return the login attribute instead of null, then we could revert this part back to user.getUsername()

Reply:
Melhor deixar getLogin.